crates/evm/src/executor/fork/cache.rs

```diff
@@ -251,10 +251,16 @@ impl MemDb {
         let mut storage = self.storage.write();
         let mut accounts = self.accounts.write();
         for (add, mut acc) in changes {
-            if acc.is_empty() || acc.is_selfdestructed() {
-                accounts.remove(&add);
-                storage.remove(&add);
+            if !acc.is_touched() {
```

Comment:
This should stay, it is a small improvement as filtering is not done inside revm and all loaded data is returned from revm.